src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -567,6 +609,57 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 		}
 	}
 
+	/// <summary>
+	/// Validate SSL validation has
+	/// </summary>
+	/// <param name="validationHash">received validation hash</param>
+	/// <param name="challenge">initial seed</param>
+	/// <param name="password">password</param>
+	/// <returns>true if validated</returns>
+	private bool ValidateFingerPrint(byte[]? validationHash, ReadOnlySpan<byte> challenge, string password)
+	{
+		if (validationHash is null || validationHash.Length == 0) return false;
+
+		// ensure using SHA256 encryption
+		if (validationHash[0] != 0x01)
+			throw new FormatException($"Unexpected validation hash format. expected 0x01 but got 0x{validationHash[0]:X2}");
+
+		byte[] passwordHashResult;
+		switch (m_pluginName)
+		{
+			case "mysql_native_password":
+				passwordHashResult = AuthenticationUtility.HashPassword(challenge, password, false);
+				break;
+
+			case "client_ed25519":
+				AuthenticationPlugins.TryGetPlugin("client_ed25519", out var ed25519Plugin);
+				passwordHashResult = ed25519Plugin!.CreatePasswordHash(password, challenge);
+				break;
+
+			default:
+				return false;
+		}
+
+		Span<byte> combined = stackalloc byte[32 + (challenge.Length - 1) + passwordHashResult.Length];
+		passwordHashResult.CopyTo(combined);
+		challenge.CopyTo(combined[passwordHashResult.Length..]);
+		m_sha2Thumbprint!.CopyTo(combined[(passwordHashResult.Length + challenge.Length - 1)..]);
+
+		byte[] hashBytes;
+#if NET5_0_OR_GREATER
+		hashBytes = SHA256.HashData(combined);
+#else
+		using (var sha256 = SHA256.Create())
+		{
+			hashBytes = sha256.ComputeHash(combined.ToArray());
+		}
+#endif
+
+		var clientGeneratedHash = hashBytes.Aggregate(string.Empty, (str, hashByte) => str + hashByte.ToString("X2", CultureInfo.InvariantCulture));
```

Comment:
This is a very inefficient way to convert a byte array to a hex string. (I'd probably go the other way and convert the server hash to a byte array, then compare.)